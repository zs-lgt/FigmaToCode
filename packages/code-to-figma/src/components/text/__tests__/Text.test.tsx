import * as React from 'react';
const renderer = require('react-test-renderer');
import { Text } from '../Text';
import { createFigma } from 'figma-api-stub';

describe('<Text />', () => {
    beforeEach(() => {
        // @ts-ignore
        global.figma = createFigma({
            simulateErrors: true
        });
    });

    it('Text without props', () => {
        const tree = renderer.create(<Text />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Text with fontFamily', () => {
        const tree = renderer.create(<Text style={{ fontFamily: 'Helvetica' }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Text with fontFamily, fontWeight', () => {
        const tree = renderer.create(<Text style={{ fontFamily: 'Helvetica', fontWeight: 'bold' }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Text with fontFamily, fontWeight, fontStyle', () => {
        const tree = renderer
            .create(<Text style={{ fontFamily: 'Helvetica', fontWeight: 'bold', fontStyle: 'italic' }} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Text with maxWidth', () => {
        const tree = renderer.create(<Text style={{ maxWidth: '100%' }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Text with width', () => {
        const tree = renderer.create(<Text style={{ width: 200 }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Text got full width when display: block', () => {
        process.env.REACT_FIGMA_WEB_DEFAULTS_ENABLED = '1';
        const tree = renderer.create(<Text style={{ display: 'block' } as any} />).toJSON();
        expect(tree).toMatchSnapshot();
        delete process.env.REACT_FIGMA_WEB_DEFAULTS_ENABLED;
    });

    it('Text got full width when display: block (wrapped to array)', () => {
        process.env.REACT_FIGMA_WEB_DEFAULTS_ENABLED = '1';
        const tree = renderer.create(<Text style={[{ display: 'block' } as any]} />).toJSON();
        expect(tree).toMatchSnapshot();
        delete process.env.REACT_FIGMA_WEB_DEFAULTS_ENABLED;
    });

    it('Text with hyperlink', () => {
        const tree = renderer
            .create(
                <Text
                    style={{ color: '#308eff' }}
                    hyperlink={{
                        type: 'URL',
                        value: 'https://react-figma.dev/'
                    }}>
                    React Figma website
                </Text>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
