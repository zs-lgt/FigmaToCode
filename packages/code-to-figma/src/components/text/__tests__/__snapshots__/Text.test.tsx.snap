// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Text got full width when display: block (wrapped to array) 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Inter",
      "style": "Regular",
    }
  }
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "display": "block",
      "minWidth": "100%",
    }
  }
/>
`;

exports[`<Text /> Text got full width when display: block 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Inter",
      "style": "Regular",
    }
  }
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "display": "block",
      "minWidth": "100%",
    }
  }
/>
`;

exports[`<Text /> Text with fontFamily 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Helvetica",
      "style": "Regular",
    }
  }
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "fontFamily": "Helvetica",
    }
  }
/>
`;

exports[`<Text /> Text with fontFamily, fontWeight 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Helvetica",
      "style": "Bold",
    }
  }
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "fontFamily": "Helvetica",
      "fontWeight": "bold",
    }
  }
/>
`;

exports[`<Text /> Text with fontFamily, fontWeight, fontStyle 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Helvetica",
      "style": "Bold Italic",
    }
  }
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "fontFamily": "Helvetica",
      "fontStyle": "italic",
      "fontWeight": "bold",
    }
  }
/>
`;

exports[`<Text /> Text with hyperlink 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 1,
          "g": 0.5568627450980392,
          "r": 0.18823529411764706,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Inter",
      "style": "Regular",
    }
  }
  hyperlink={
    Object {
      "type": "URL",
      "value": "https://react-figma.dev/",
    }
  }
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "color": "#308eff",
    }
  }
>
  React Figma website
</text>
`;

exports[`<Text /> Text with maxWidth 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Inter",
      "style": "Regular",
    }
  }
  hasDefinedWidth="100%"
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "maxWidth": "100%",
    }
  }
/>
`;

exports[`<Text /> Text with width 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Inter",
      "style": "Regular",
    }
  }
  hasDefinedWidth={200}
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={
    Object {
      "width": 200,
    }
  }
  width={200}
/>
`;

exports[`<Text /> Text without props 1`] = `
<text
  fills={
    Array [
      Object {
        "color": Object {
          "b": 0,
          "g": 0,
          "r": 0,
        },
        "opacity": 1,
        "type": "SOLID",
      },
    ]
  }
  fontName={
    Object {
      "family": "Inter",
      "style": "Regular",
    }
  }
  innerRef={
    Object {
      "current": undefined,
    }
  }
  loadedFont={null}
  style={Object {}}
/>
`;
